src/serializer.ts

```diff
@@ -84,7 +84,7 @@ export class Serializer {
      * @param clazz The class constructor.
      * @returns An instance of the type `T` with the prototype of `clazz` or one of its registered children.
      */
-    public deserialize<T>(obj: any, clazz?: any): T {
+    public deserialize<T>(obj: any, clazz?: any /* TODO See if we can change this any */): T {
```

Comment:
The issue with this is that an `abstract class` is not `Instantiable` because you can't call `new` on it. That's the only reason why there's an `any` here.